spec/spec-0.2.md

```diff
@@ -0,0 +1,1020 @@
+| Remote        | **required** | --              | -- (URI is used)             | optional |
+| Remote        | **required** | --              | absolute URI (overrides URI) | optional |
+
+
+
+By defining a mapping of URIs to names, the OCIF file becomes more readable and easier to maintain.
+
+**Example**  
+A schema array with two schemas: 
+```json 
+{
+  "schemas": [
+    {
+      "uri": "https://spec.canvasprotocol.com/node/ports/0.2",
+      "name": "@ocwg/node/ports"
+    },
+    {
+      "uri": "https://example.com/ns/ocif-node/circle/1.0",
+      "location": "schemas/circle.json",
+      "name": "@example/circle"
+    }
+  ]
+}
+```
+
+### Built-in Schema Mappings
+To simplify the use of OCIF, a set of built-in schema mappings is defined.
+
+Any [Schema Name](#schema-name) of the form  
+`@ocwg/rel/`_suffix_ maps to a schema [URI](#uri)  
+`https://spec.canvasprotocol.org/rel/` _suffix_ `/0.2` 
+
+Here `0.2` is the current version of the OCIF spec. Later OCIF specs will have different versions and thus different URIs.
+
+Built-in Entries, where the syntax `{var}` denotes placeholders.
+```json
+{
+  "schemas": [
+    { "name": "@ocwg/node/{ext-type}", "uri": "https://spec.canvasprotocol.com/node/{ext-type}/0.2" },
+    { "name": "@ocwg/rel/{ext-type}", "uri": "https://spec.canvasprotocol.com/rel/{ext-type}/0.2" }
+  ]
+}
+```
+
+
+
+These mappings SHOULD be materialized into the OCIF JSON schema.
+
+# Extensions
+No two canvas applications are alike: 
+There are apps for informal whiteboarding, formal diagramming, quick visual sketches, node-and-wire programming, and many other use cases. 
+Each of these apps has radically different feature sets. 
+Extensions are an integral part of OCIF. 
+They allow adding custom data to nodes, relations, and resources.
+
+- An extension is a JSON object (used as a "property bag") with one mandatory property: `type`.
+Thus, `type` is a reserved property key. 
+All other property keys can be used by the extension.
+- Arbitrary, nested JSON structures are allowed.
+- Extensions SHOULD define how the base properties play together with the extension properties and with other (known) extensions.
+- Nodes and relations can have multiple extensions within their `data` array.
+- Each extension is an object with a `type` property.
+
+| Property | JSON Type | OCIF Type                                  | Required     | Contents          |
+|----------|-----------|:-------------------------------------------|--------------|-------------------|
+| `type`   | `string`  | [Schema Name](#schema-name) or [URI](#uri) | **required** | Type of extension |
+
+
+- **type**: The type of the extension. This is a URI or a simple name.
+  If a name is used, that name must be present in the [schemas](#schemas) section, where it is mapped to a URI.
+
+
+For an example of an extension, see the [appendix](#appendix), [Node Extension: Circle](#node-extension-circle).
+
+## Defining Extensions
```

Comment:
I expected to see a table here rather than the bulleted list of recommendations. Was that omitted on purpose?